sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -1271,6 +1271,10 @@ ConnectionPolicy buildConnectionPolicy() {
     private void validateConfig() {
         URI uri;
         try {
+            // split the serviceEndpoint in case of a connection string being passed in
```

Comment:
Can we check and validate each part of the connection string first?

Reply:
I could validate first but I think it would be beneficial to the customer if we accept endpoints that are slightly off for example localhost:8080 where it would fail validation, but we just need to add the https://. Accepting these would be new, so I am fine changing it to have validation. What are your thoughts on this?

Reply:
Changed to only reconstructing when path or query is specified for the uri to minimize risk of breaking changes and keep logic simple.